### PR TITLE
[root] Add version 6.22

### PR DESCRIPTION
**Pull request by HadrienG2**
From a build system point of view, this was a relatively tame release. However, the xextproto issue from <=6.08 is back in this release.

Here I'm doing the minimal step of enabling the xextproto include path hack for >=6.22. Alternate options are to...

- Enable it for every ROOT version. It does not really harm, even though it's not always necessary.
- Cross-check if current version of spack can live without the evil unsetting of the SPACK_INCLUDE_DIRS environment variable altogether, freeing us from all those include path hacks.
    * AFAIK, spack's usage of -I / -isystem has changed recently, and it was the original cause of all those include path problems. So with recent spack versions, the root package's convoluted manual include path setup may not be necessary anymore.
    * To check this, I need to know in which circumstances those include paths problems would manifest before those tricks were introduced. I think @gartung was the author of the original bugfix that unset SPACK_INCLUDE_DIRS, so he might remember a failing test scenario that I can exercise.

### CODE REVIEW

**Comment by gartung** [comment]
As I recall this was done because spack inserts the -I flags before the -isystem flags that ROOT was using. This caused some header to get the wrong precedence which broke the build. @chissg might recall the details.

**Comment by gartung** [comment]
I think maybe it was the python header.

**Comment by HadrienG2** [comment]
Oh, silly me, I actually wrote it down myself: https://github.com/spack/spack/pull/17459/files#diff-09e58717bcd0dfb20a2714fdeaa39ec1R456 .

Time to get experimenting...

**Comment by HadrienG2** [comment]
Turns out I do not have the time to investigate this right now, so I propose to merge this as-is to unblock people who are waiting for ROOT 6.22 in Spack.

**Comment by gartung** [comment]
Needs at least on review before I can merge it.

**Review by tgamblin — APPROVED** [review]

**Pull request merged**
